DSharpPlus/Entities/Channel/DiscordThreadChannelMember.cs

```diff
@@ -0,0 +1,124 @@
+// This file is part of the DSharpPlus project.
+//
+// Copyright (c) 2015 Mike Santiago
+// Copyright (c) 2016-2021 DSharpPlus Contributors
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DSharpPlus.Exceptions;
+using DSharpPlus.Net.Abstractions;
+using DSharpPlus.Net.Models;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities
+{
+    /// <summary>
+    /// Represents a discord thread member object.
+    /// </summary>
+    public class DiscordThreadChannelMember : IEquatable<DiscordThreadChannelMember>
+    {
+        /// <summary>
+        /// Gets ID of the thread.
+        /// </summary>
+        [JsonProperty("id", NullValueHandling = NullValueHandling.Ignore)]
+        public ulong Id { get; internal set; }
+
+        /// <summary>
+        /// Gets ID of the user.
+        /// </summary>
+        [JsonProperty("user_id", NullValueHandling = NullValueHandling.Ignore)]
+        public ulong UserId { get; internal set; }
+
+        /// <summary>
+        /// Gets timestamp when the user joined the thread.
+        /// </summary>
+        [JsonIgnore]
+        public DateTimeOffset? JoinTimeStamp
+            => !string.IsNullOrWhiteSpace(this.JoinTimeStampRaw) && DateTimeOffset.TryParse(this.JoinTimeStampRaw, CultureInfo.InvariantCulture, DateTimeStyles.None, out var dto) ?
+                dto : null;
+
+        [JsonProperty("join_timestamp", NullValueHandling = NullValueHandling.Ignore)]
+        internal string JoinTimeStampRaw { get; set; }
+
+        /// <summary>
+        /// Gets flags (Currently only for notification settings).
+        /// </summary>
+        [JsonProperty("flags", NullValueHandling = NullValueHandling.Ignore)]
```

Comment:
I've yet to read on the documentation, but I think this should be an enum?

Reply:
It's an integer atm an no documentation about the flags directly.
So I would let it be a int for now

Reply:
![image](https://user-images.githubusercontent.com/14029133/119247728-908e6a00-bb8c-11eb-980d-011b3e7b63fd.png)

Reply:
I'll do a bit reverse engineering on client side, maybe I find something out to create a Flag enum

Reply:
3: All
5: Only @mentions
9: Nothing